run_dir/static/js/project_samples.js

```diff
@@ -653,10 +653,283 @@ function load_samples_table(colOrder) {
+                            'href="' + lims_uri + '/clarity/search?scope=Sample&query='+info[column_id]+'">'+
+                            info[column_id] + '</a></td>';
+                          }
+                  }
+
+                // Sample run metrics is an array of links - link to flowcells page
+                else if (column_id == 'sample_run_metrics') {
+                  tbl_row += '<td class="' + column_id + '">';
+                  for (var i=0; i<info[column_id].length; i++) {
+                    var fc = info[column_id][i];
+                    // Remove the lane number and barcode - eg 6_FCID_GTGAAA
+                    fc = fc.substring(2);
+                    fc = fc.replace(/_[ACTG\-]+$/,'');
+                    fc = fc.replace('_NoIndex', '');
+                    tbl_row += '<samp class="nowrap"><a class="text-decoration-none" href="/flowcells/' + fc + '">' +
+                    info[column_id][i] + '</a></samp><br>';
+                  }
+                  tbl_row += '</td>';
+                }
+
+                // Library prep is an array of *Objects*
+                else if (column_id == 'library_prep') {
+                  tbl_row += '<td class="' + column_id + '">';
+                  tbl_row += auto_format(prep, true) + ' </td>';
+                }
+
+                // Make sure that 'million reads' has two decimal places
+                else if (column_id == 'total_reads_(m)' && typeof info[column_id] !== 'undefined'){
+                   tbl_row += '<td class="' + column_id + ' text-right">' + Number(info[column_id]).toFixed(2) + '</td>';
+                }
+
+                // Split the preps
+                else if (column_id == 'prep_status'){
+                  tbl_row += auto_samples_cell(column_id, prepinfo[column_id]);
+                }
+
+                // everything else
+                else {
+                  tbl_row += auto_samples_cell(column_id, info[column_id]);
+                }
+
+            }
+            else if (value[2] == "initial-qc-columns" && info['initial_qc'] !== undefined) {
+                info['initial_qc'][column_id] = round_floats(info['initial_qc'][column_id], 2);
+
+                // Fragment Analyzer Image
+                if (column_id == 'frag_an_image'){
+                    tbl_row += '<td class="' + column_id + '">'+
+                              '<span class="caliper_loading_spinner">'+
+                                '<span class="fa fa-sync fa-spin"></span>  Loading image..</span>'+
+                              '</span>'+
+                              '<a id="caliper_thumbnail_'+info['scilife_name']+'" class="caliper-thumbnail loading" href="'+info['initial_qc'][column_id]+'" data-imgtype="Initial QC Fragment Analyzer Image" data-samplename="'+info['scilife_name']+'"></a>'+
+                            '</td>';
+                }
+
+                // Caliper image
+                else if (column_id == 'caliper_image'){
```

Comment:
Theres no caliper in the library validation view, is there?

Reply:
True! I just copied everything :-)